app/actions/tests/test_pull_events.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+from app.conftest import async_return
+from app.services.action_runner import execute_action
+
+
+@pytest.mark.asyncio
+async def test_execute_pull_observations_action(
+        mocker, mock_gundi_client_v2, mock_state_manager, inaturalist_integration_v2,
+        mock_get_gundi_api_key, mock_gundi_sensors_client_class,
+        mock_get_observations_v2, mock_publish_event, mock_gundi_client_v2_class
+):
+    mock_gundi_client_v2.get_integration_details.return_value = async_return(inaturalist_integration_v2)
+    mocker.patch("app.services.action_runner._portal", mock_gundi_client_v2)
+    mocker.patch("app.services.activity_logger.publish_event", mock_publish_event)
+    mocker.patch("app.services.action_runner.publish_event", mock_publish_event)
+    mocker.patch("app.actions.handlers.state_manager", mock_state_manager)
+    mocker.patch("app.services.gundi.GundiClient", mock_gundi_client_v2_class)
+    mocker.patch("app.services.gundi.GundiDataSenderClient", mock_gundi_sensors_client_class)
+    mocker.patch("app.services.gundi._get_gundi_api_key", mock_get_gundi_api_key)
+    mocker.patch("app.actions.handlers.get_observations_v2", mock_get_observations_v2)
+
+    response = await execute_action(
+        integration_id=str(inaturalist_integration_v2.id),
+        action_id="pull_events"
+    )
+    assert "result" in response
+    assert response["result"].get("events_extracted")
```

Comment:
we could assert that the number of extracted events matches what is expected according to the mock data, and probably the same for the photos.